nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ControllerServiceResource.java

```diff
@@ -464,6 +464,10 @@ public Response updateControllerServiceReferences(
                     required = true
             ) final UpdateControllerServiceReferenceRequestEntity requestUpdateReferenceRequest) {
 
+        if (requestUpdateReferenceRequest == null) {
+            throw new IllegalArgumentException("The controller service information must be specified.");
+        }
+
         if (requestUpdateReferenceRequest.getId() == null) {
```

Comment:
You can probably club the conditional here

```
if(requestUpdateReferenceRequest == null || requestUpdateReferenceRequest.getId() == null)
```

Reply:
Okay. Thanks.